packages/vscode-extension/src/commands/CompareFile.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { UNRESTRICTED_FOLDERS } from '../ApplicationConstants';
+import { COMPARE_FILE, IMPORT_FILES } from '../service/TranslationKeys';
+import { actionResultStatus, ApplicationConstants } from '../util/ExtensionUtil';
+import FileImportCommon from './FileImportCommon';
+
+export default class CompareFile extends FileImportCommon {
+	private static readonly COMMAND_NAME = 'comparefile';
+    private static readonly TEMP_FOLDER_PREFIX = 'suitecloud-vscode-extension-compare-file-';
+
+	constructor() {
+		super(CompareFile.COMMAND_NAME);
+	}
+
+	protected validateBeforeExecute() {
+		const superValidation = super.validateBeforeExecute();
+		if (!superValidation.valid) {
+			return superValidation;
+		}
+
+		if (!this.activeFileIsUnderUnrestrictedFolder()) {
+			return this.unsuccessfulValidation(this.translationService.getMessage(IMPORT_FILES.ERROR.NOT_ALLOWED_FOLDER));
+		}
+
+		return this.successfulValidation();
+	}
+
+	protected async execute() {
+		const activeFilePath = this.activeFile!;
+		const tempFolderPath = fs.mkdtempSync(path.join(os.tmpdir(), CompareFile.TEMP_FOLDER_PREFIX));
+
+		this.copyManifestFileToTempFolder(tempFolderPath);
```

Comment:
why do we need to copy this?

Reply:
The temporary directory becomes a "clone" of the project. That's where the import files is actually being performed. So that means project.json is also needed to do the actual import.

Reply:
same as above.